contracts/src/defuse/verify/diff/tokens.rs

```diff
@@ -0,0 +1,91 @@
+use std::collections::btree_map::{self, BTreeMap, Entry};
+
+use impl_tools::autoimpl;
+use near_sdk::near;
+use serde_with::{serde_as, DisplayFromStr};
+
+use crate::defuse::{tokens::TokenId, DefuseError};
+
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(
+    all(feature = "abi", not(target_arch = "wasm32")),
+    serde_as(schemars = true)
+)]
+#[cfg_attr(
+    not(all(feature = "abi", not(target_arch = "wasm32"))),
+    serde_as(schemars = false)
+)]
+#[near(serializers = [borsh, json])]
+#[autoimpl(Deref using self.0)]
+pub struct TokenDeltas(
+    /// [`BTreeMap`] ensures deterministic order
```

Comment:
Why is it so important for us?

Reply:
Deterministic order during serialization is needed for signature verification, since the we sign the hash of the payload serialized to borsh.